### PR TITLE
DPDK Refactor: parallel util, use node.nics, dpdk sample apps

### PR DESCRIPTION
**Pull request by mcgov**
Adding the multiprocess test required compiling dpdk sample apps.
This PR refactors a few things in the dpdksuite to use the new utils and tools available.

- Allow testpmd tool to compile sample apps
- use parallel util to run concurrent items
- use node.nics instead of directly instantiating the Nics class
- expose dpdk build path as variable in testpmd to allow calling of sample apps later

### CODE REVIEW

**Comment by mcgov** [comment]
note: depends on https://github.com/microsoft/lisa/pull/1790, will need to rebase once those merge.

**Review by squirrelsc — DISMISSED** [review]

**Comment by squirrelsc** [comment]
If it works, can we merge it firstly?

**Comment by mcgov** [comment]
Should merge clean after we merged the dependencies! Hopefully don't need a second approval. EDIT: Nope. Needed the rebase still.

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**
